Core/Services/CarService.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Net;
+using System.Threading.Tasks;
+using AutoMapper;
+using Core.DTO.CarDTO;
+using Core.Entities.CarCategoryEntity;
+using Core.Entities.CarEntity;
+using Core.Entities.UserEntity;
+using Core.Exceptions;
+using Core.Interfaces;
+using Core.Interfaces.CustomService;
+using Core.Resources;
+using Microsoft.AspNetCore.Identity;
+
+namespace Core.Services
+{
+    public class CarService : ICarService
+    {
+        private readonly UserManager<User> _userManager;
+        private readonly IRepository<Car> _carRepository;
+        private readonly IRepository<CarCategory> _categoryRepository;
+        private readonly IMapper _mapper;
+
+        public CarService(
+            IRepository<Car> car,
+            IRepository<CarCategory> categoryRepository,
+            IMapper mapper,
+            UserManager<User> userManager)
+        {
+            _carRepository = car;
+            _categoryRepository = categoryRepository;
+            _mapper = mapper;
+            _userManager = userManager;
+        }
+        public async Task<CarDTO> AddCarAsync(CreateCarDTO createCarDTO, string userId)
+        {
+            var car = _mapper.Map<Car>(createCarDTO);
+            car.CreationDate = DateTimeOffset.UtcNow;
+            car.Category = await _categoryRepository.GetByIdAsync(createCarDTO.CategoryId);
```

Comment:
You can set the category id only

Reply:
I need the whole category object because in DTO I return not categoryId, but it's name.